Add Github Actions to automatic deployment of the wiki.

manuelvsousa opened the pull request:
The followed configuration process was [this one](https://github.com/peaceiris/actions-gh-pages) for github pages and [this one](https://github.com/peaceiris/actions-hugo) for hugo stuff. A deploy key was added in the org secrets, and everything should be automatic. We can *later* add scripts to verify the wiki's integrity.
All PR's and pushes to master will be verified whether hugo is able to run: mostly it verifies if there are no broken links, etc. 

This feature is particularly important once we start accepting community contributions. 

**Important note to future maintainers:** I won't save the private key, so if the time comes, you should create a new deploy key and ignore the current one which is only useful for the deployment process.

empijei commented:
Could you please write instructions on how to do what you describe here with the keys?

empijei commented on .github/workflows/pages.yml:
```diff
@@ -0,0 +1,33 @@
+
+jobs:
+  deploy:
+    runs-on: ubuntu-20.04
```
do we have to pin a version? can't we use just ubuntu latest?

manuelvsousa replied:
I changed it to ubuntu-latest

manuelvsousa commented:
> Could you please write instructions on how to do what you describe here with the keys?

It's done. I added the "plugins" we use, but I don't specifically say what is needed to set the keys. I added the chosen strategy and an hyperlink to instructions in the original repository which has incredible documentation about it.
